### PR TITLE
You can now stuff people on the floor into pet carriers

### PR DESCRIPTION
**Pull request by silicons**
Takes 20 seconds to break out, and breaking out will hard stun them for a while so this isn't a "hide in locker" meme.

### CODE REVIEW

**Comment by Poojawa** [comment]
Goodness Kev, your kinkiness is showing again. which fox girl has been threatening you with the crate?

File: code/game/objects/items/pet_carrier.dm [hunk 1]
```diff
@@ -86,19 +89,25 @@
 	if(!open)
 		to_chat(user, "<span class='warning'>You need to open [src]'s door!</span>")
 		return
+	if(user == target)
+		to_chat(user, "<span class='warning'>Why would you ever do that?</span>")
+		return
```
Comment (deathride58) [inline code/game/objects/items/pet_carrier.dm]: No fun allowed??? :(

File: code/game/objects/items/pet_carrier.dm [hunk 2]
```diff
@@ -89,16 +92,19 @@
 	if(target.mob_size > max_occupant_weight)
 		if(ishuman(target))
```
Comment (Ghommie) [inline code/game/objects/items/pet_carrier.dm:93]: This feature could honestly also be expanded to monkeys, all it has be done is change a couple is_type checks, and adjust the to_chat messages at line 99.

File: code/game/objects/items/pet_carrier.dm [hunk 3]
```diff
@@ -89,16 +92,19 @@
...
+			if(!allow_humans)
+				if(iscatperson(H))
+					to_chat(user, "<span class='warning'>You'd need a lot of catnip and treats, plus maybe a laser pointer, for that to work.</span>")
+				else
+					to_chat(user, "<span class='warning'>Humans, generally, do not fit into pet carriers.</span>")
 			else
-				to_chat(user, "<span class='warning'>Humans, generally, do not fit into pet carriers.</span>")
+				if(H.resting)
+					user.visible_message("<span class='boldwarning'>[user] starts stuffing [H] into [src]!</span>")
+					if(!do_after(user, 50, target = H))
```
Comment (Ghommie) [inline code/game/objects/items/pet_carrier.dm:103]:
```suggestion
					if(!do_after(user, 50, target = H) || !H.resting)
```

You'd have to rewrite this line anyway if you follow up my suggestion above.

File: code/game/objects/items/pet_carrier.dm [hunk 4]
```diff
@@ -89,16 +92,19 @@
 	if(target.mob_size > max_occupant_weight)
 		if(ishuman(target))
 			var/mob/living/carbon/human/H = target
```
Comment (Ghommie) [inline code/game/objects/items/pet_carrier.dm:94]: This line is unnecessary, resting is a living var and the iscatperson() define, checks if the mob is human before hand.
So no phantom issues with the above suggestion, I believe.
Though you'll probably ignore this as not a current addition.

**Pull request closed without merging**